crates/red_knot_python_semantic/src/types/infer.rs

```diff
@@ -1644,7 +1648,7 @@ impl<'db> TypeInferenceBuilder<'db> {
             }
         }
 
-        if let Some(value) = value {
+        if let Some(value) = assignment.value.as_deref() {
```

Comment:
Why the change to use `assignment.value` here and no longer unpack `value` above?

Reply:
Same as above